features/security/src/main/feature/feature.xml

```diff
@@ -21,694 +21,317 @@
-        <feature>platform-error-page-injector</feature>
-        <feature>war</feature>
-    </feature>
+        <!-- Expansion -->
+        <bundle>mvn:ddf.security.expansion/security-expansion-api/${project.version}</bundle>
 
-    <feature name="session-invalidator" version="${project.version}"
-             description="API for invalidating servlet sessions">
-        <bundle>mvn:ddf.platform/platform-session-invalidator/${project.version}</bundle>
-    </feature>
+        <!-- Token Storage -->
+        <bundle>mvn:ddf.security.storage/token-storage-api/${project.version}</bundle>
 
-    <feature name="platform-paxweb-jettyconfig" version="${project.version}"
-      description="Custom Session Manager">
-        <config name="org.ops4j.pax.web" append="true">
-#######################################
-# Jetty Server Authenticator Settings
-#######################################
+        <!-- Sessions -->
+        <bundle>mvn:ddf.platform.security/session-management-api/${project.version}</bundle>
+        <bundle>mvn:ddf.platform/platform-session-invalidator-api/${project.version}</bundle>
 
-# Set the authmethod/realmname so that our JettyAuthenticator is used
-org.ops4j.pax.web.default.authmethod=DDF
-org.ops4j.pax.web.default.realmname=DDF
-        </config>
-        <feature version="${spring.version}_1">spring</feature>
-        <feature>apache-commons</feature>
-        <feature>security-core-api</feature>
-        <feature>security-core-impl</feature>
-        <feature>security-core-services</feature>
-        <feature>security-filter-api</feature>
-        <feature>pax-http-jetty</feature>
-        <bundle start-level="30">mvn:org.ops4j.pax.web/pax-web-jsp/${pax.web.version}</bundle>
-        <bundle>mvn:javax.el/javax.el-api/3.0.0</bundle>
-        <feature>pax-jetty</feature>
-        <feature>session-invalidator</feature>
-        <feature prerequisite="true">wrap</feature>
-        <bundle>wrap:mvn:net.jodah/failsafe/${jodah-failsafe.version}</bundle>
-        <feature>guava</feature>
-        <feature>common-system</feature>
-        <bundle>mvn:ddf.platform/platform-paxweb-jettyconfig/${project.version}</bundle>
-    </feature>
+        <!-- Logout -->
+        <bundle>mvn:ddf.security.servlet/security-servlet-logout-api/${project.version}</bundle>
 
-    <feature name="security-handler-api" version="${project.version}"
-             description="API for authentication handlers for web applications.">
-        <feature>security-core-api</feature>
-        <feature>parser-xml</feature>
-        <feature>oidc-dependencies</feature>
-        <bundle>mvn:org.apache.cxf/cxf-rt-rs-security-jose/${cxf.version}</bundle>
-        <bundle>mvn:org.apache.cxf/cxf-rt-rs-security-jose-jaxrs/${cxf.version}</bundle>
-        <bundle>mvn:ddf.security.handler/security-handler-api/${project.version}</bundle>
-    </feature>
+        <!-- Email -->
+        <bundle>mvn:ddf.platform.email/platform-email-api/${project.version}</bundle>
```

Comment:
Oman this blew up.

I too thought it was weird that there were platform packages in here. They do depend on security packages, but I decided moved them over to the `utilities` feature file anyway.

But when I did that, the `utilities` feature itests they claimed they couldn't find the `org.codice.ddf.platform.util` package, even though I added the `platform-util` feature which exposes that package. No matter what I did, the email bundles just were not picking up that package.

The only way I could fix it was by including the big `security-core-services` feature. I have a feeling its because in `platform-util`'s bundle manifest the `org.codice.ddf.platform.util` exported package "uses" a ton of other packages that `security-core-services` ends up pulling in.

I couldn't find an easy way to determine those packages, so I'm just going to make the platform bundles depend on `security-core-services`. I have verified that we only pull in those platform bundles when we have already pulled in `security-core-services` anyway. I don't like it, but I also don't feel like doing an insane amount of trial and error or reading up on package resolution in karaf.

Reply:
Huh, that is weird that it couldn't find `org.codice.ddf.platform.util` with that feature installed. platform-email-impl pulls in the security logger service, though, so it does depend on `security-core-services`

Reply:
Oh I think I know what the issue _could_ have been.

I forgot to add the `security` feature file dependency to the `utilitlies` feature pom. When I did, maven noted a cyclical dependency. Both having a cyclical dependency and not including the `security` feature file as a dependency could have caused the weird behavior I observed.

TLDR: I'm keeping the email bundles in place or else it introduces a cyclical dependency.